uicomponent-compose/main/src/main/java/io/github/droidkaigi/feeder/viewmodel/RealTimetableDetailViewModel.kt

```diff
@@ -0,0 +1,80 @@
+package io.github.droidkaigi.feeder.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import io.github.droidkaigi.feeder.AppError
+import io.github.droidkaigi.feeder.TimetableContents
+import io.github.droidkaigi.feeder.repository.TimetableRepository
+import io.github.droidkaigi.feeder.timetable2021.TimetableDetailViewModel
+import javax.annotation.meta.Exhaustive
+import javax.inject.Inject
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class RealTimetableDetailViewModel @Inject constructor(
+    private val repository: TimetableRepository,
+) : ViewModel(), TimetableDetailViewModel {
+
+    private val effectChannel = Channel<TimetableDetailViewModel.Effect>(Channel.UNLIMITED)
+    override val effect: Flow<TimetableDetailViewModel.Effect> = effectChannel.receiveAsFlow()
+
+    init {
+        viewModelScope.launch {
+            refreshRepository()
+        }
```

Comment:
This process is required for opening the Detail directly and for test cases..

Reply:
How about adding some comments like "// For cases where the screen is opened directly by deeplink or for testing." 👍

Reply:
Good idea!
I'll add it, thanks!